app/src/server/api/routers/kage.ts

```diff
@@ -187,6 +187,9 @@ export const kageRouter = createTRPCRouter({
 
       // Derived
       const structure = userVillage?.structures.find((s) => s.id === input.structureId);
+      const discountLevel =
```

Comment:
I'm thinking for consistency maybe we should move this into `calcStructureUpgrade` so that it's calculated the same place as the tax, and returned from that function alongside the tax?